pkg/cnab/dependencies_v2.go

```diff
@@ -0,0 +1,82 @@
+package cnab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	v2 "get.porter.sh/porter/pkg/cnab/extensions/dependencies/v2"
+)
+
+const (
+	// DependenciesV2ExtensionShortHand is the short suffix of the DependenciesV2ExtensionKey
+	DependenciesV2ExtensionShortHand = "dependencies@v2"
+
+	// DependenciesV2ExtensionKey represents the full key for the DependenciesV2Extension.
+	DependenciesV2ExtensionKey = "org.getporter." + DependenciesV2ExtensionShortHand
+
+	// DependenciesV2Schema represents the schema for the DependenciesV2 Extension
+	DependenciesV2Schema = "https://getporter.org/extensions/dependencies/v2/schema.json"
+)
+
+// DependenciesV2Extension represents the required extension to enable dependencies
+var DependenciesV2Extension = RequiredExtension{
```

Comment:
nit: Definitely not in love with the naming of these structures/methods having "v2" directly in them. Would it be worthwhile to create a dependencies interface and use more like the factory pattern?

Reply:
That would involve a larger change to how we define and process CNAB extensions which is out of scope for this issue. I am not certain that I see a way for it to work since there isn't a consistent interface for extensions, they return a unique data structure representing the parsed extension data from the custom section of bundle.json. So the docker extension returns a different parsed structure than the v1 deps extension, than the v2 deps extension, etc.

The use of the extension you see defined here is limited to this package. Elsewhere when you want to read out the dependencies v2 data structure from the bundle, you can do this:

```go
var bundle cnab.ExtendedBundle // initialize this somehow
if bundle.HasDependenciesV2() {
  depsDef := bundle.ReadDependenciesV2()
  // ...
}
```

If you want to take a stab at it anyway, I recommend a follow-up draft PR showing how you'd like to see it changed since I'm just not seeing it since I'm a bit too used to how it's implemented today.